File: CodingDojo/bomberman-servers/cpp/bomberman/utils/Point.cpp
#include "Point.h"


Point::Point(int _x, int _y) : x(_x), y(_y), isSetAsNull(false) {
}

bool Point::operator==(const Point& pt) const {
	return equals(pt);
}

bool Point::equals(const Point& pt) const {
	return x == pt.x && y == pt.y;
}

bool Point::isNull() const {
	return isSetAsNull;
}

void Point::setNull(bool state) {
	isSetAsNull = state;
}

String Point::toString() const {
	StringStream ss;
	ss << "[" << x << "," << y << "]";
	return ss.str();
}

bool Point::isBad(int boardSize) const {
	return x >= boardSize || y >= boardSize || x < 0 || y < 0;
}

int Point::getX() const {
	return x;
}

int Point::getY() const {
	return y;
}

PointList Point::getSurrounds(int boardSize) const {
	PointList surr;
	for (int i = -1; i <= 1; ++i) {
		for (int j = -1; j <= 1; ++j) {
			Point tmpP(x + i, y + j);
			if (i != j && i != -j && !tmpP.isBad(boardSize) && !equals(tmpP)) {
				surr.push_back(tmpP);
			}
		}
	}
	return surr;
}

File: CodingDojo/clients/bomberman/java/src/main/java/com/utils/Point.java
package com.utils;

/**
 * User: oleksandr.baglai
 */
public class Point {
    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public Point(Point pt) {
        this(pt.x, pt.y);
    }

    public static Point pt(int x, int y) {
        return new Point(x, y);
    }

    @Override
    public boolean equals(Object o) {
        Point pt = (Point)o;
        return pt.x == x && pt.y == y;
    }

    @Override
    public String toString() {
        return String.format("[%s,%s]", x, y);
    }

    public boolean isBad(int boardSize) {
        return x >= boardSize || y >= boardSize || x < 0 || y < 0;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }
}